### ISSUE TITLE
Issues with Tacotron2 Training on Nawar Halabi's Arabic Speech Corpus

### ISSUE THREAD

**Issue by hayderkharrufa** [nipponjo/tts-arabic-pytorch#10]
Hello,

I've been attempting to train Tacotron2 using Nawar Halabi's Arabic Speech Corpus, I'm trying to make sure everything works fine with this corpus before moving on to my own data. Despite following the instructions for preprocessing and training with the provided configurations, I'm experiencing issues with the quality of the synthesized speech.

### Training Details

- Preprocessed the audio files using `scripts/preprocess_audio.py` (after fixing minor issues).
- Trained the model using `configs/nawar.yaml` configuration file.
- Tried with both `train_tc2.py` and `train_tc2_adv.py`.
- I'm using a single GeForce RTX 3090 GPU and Ubuntu 22.04.2 LTS.

#### nawar.yaml

```yaml
restore_model: ''

log_dir: logs/exp2
checkpoint_dir: checkpoints/exp2

train_wavs_path: /media/hayder/Disk2/development/tts-arabic-pytorch_test_2/data/arabic-speech-corpus/wav_new
train_labels: ./data/train_phon.txt

test_wavs_path: /media/hayder/Disk2/development/tts-arabic-pytorch_test_2/data/arabic-speech-corpus/test set/wav_new
test_labels: ./data/test_phon.txt

balanced_sampling: False
sampler_weights_file: ./data/sampler/sampler_weights

cache_dataset: False

n_save_states_iter: 10
n_save_backup_iter: 1000
```

#### basic.yaml
```yaml
# training
epochs: 500
decoder_max_step: 3000

random_seed: False

batch_size: 8
learning_rate: 1.0e-3
weight_decay: 1.0e-6
grad_clip_thresh: 1.0

cache_dataset: True
use_cuda_if_available: True

balanced_sampling: False

# vocoder
vocoder_state_path: pretrained/hifigan-asc-v1/hifigan-asc.pth
vocoder_config_path: pretrained/hifigan-asc-v1/config.json

# diacritizers
shakkala_path: pretrained/diacritizers/shakkala_second_model6.pth
shakkelha_path: pretrained/diacritizers/shakkelha_rnn_3_big_20.pth
```

### Testing Script
```python3
from models.tacotron2 import Tacotron2Wave
import soundfile as sf
import playsound

model = Tacotron2Wave('/media/hayder/Disk2/development/tts-arabic-pytorch_test_2/checkpoints/exp2/states.pth')
model = model.cuda()
wave = model.tts("اَلسَّلامُ عَلَيكُم يَا صَدِيقِي")

sf.write('audio.wav', wave, 22050)
playsound.playsound('audio.wav')
```

### Inaccurate Synthesized Output
The generated speech does not align with the expected pronunciation. I've included a WAV file example of the synthesized speech for the sentence "اَلسَّلامُ عَلَيكُم يَا صَدِيقِي" [here](https://drive.google.com/file/d/1-RJ-h27P6uyhLTPQTF_005f872rUa4h7/view?usp=sharing).

Could you please assist me in identifying the possible issues? If you need additional information I'll promptly provide it.

Thanks.

**Comment by hayderkharrufa** [comment]
I tried reverting back to the first commit and train using train.py and everything worked as expected.

**Comment by nipponjo** [comment]
Hello, thank you for making me aware of this issue and sorry for not replying earlier. There might be a problem with the loss formulation, so I have rolled it back to the old one for now. I don't know, if it's necessary, but when training the Tacotron2 model, I used the pretrained weights from [TACOTRON2_WAVERNN_CHAR_LJSPEECH](https://pytorch.org/audio/stable/generated/torchaudio.pipelines.TACOTRON2_WAVERNN_CHAR_LJSPEECH.html#tacotron2-wavernn-char-ljspeech). Of course, when you train on another Arabic dataset, it is better to use the weight from the Arabic Speech Corpus and you should get results quickly.

**Comment by hayderkharrufa** [comment]
Thank you for your previous response.

I tried training again with Nawar Halabi's Arabic Speech Corpus using both train_tc2.py and train_tc2_adv.py as provided in the current version of the repo, without providing starting weights. The issue still persists, and I'm unable to get good results with either script.

When I revert to the first commit of the repo and start training using train.py (without starting weights), everything works as expected.

I hope this information helps in diagnosing the problem.
